cmake/compiler_support.cmake

```diff
@@ -0,0 +1,261 @@
+###############################################################################
+#
+# (c) Copyright IBM Corp. 2017, 2017
+#
+#  This program and the accompanying materials are made available
+#  under the terms of the Eclipse Public License v1.0 and
+#  Apache License v2.0 which accompanies this distribution.
+#
+#      The Eclipse Public License is available at
+#      http://www.eclipse.org/legal/epl-v10.html
+#
+#      The Apache License v2.0 is available at
+#      http://www.opensource.org/licenses/apache2.0.php
+#
+# Contributors:
+#    Multiple authors (IBM Corp.) - initial implementation and documentation
+###############################################################################
+
+# This file contains a number of support pieces required to build the compiler
+# component. 
+# 
+# One of the building blocks of this support is the notion that each compiler
+# you may want to build has an associated 'name'. For example jitbuilder is 
+# the name of one. 
+
+## Tools and helpers:
+
+Find_package(Perl)
+
+if (NOT PERL_FOUND )
+	message(FATAL_ERROR "Perl not found")
+endif()
+
+set(MASM2GAS_PATH ${OMR_ROOT}/tools/compiler/scripts/masm2gas.pl CACHE INTERNAL "MASM2GAS PATH")
+
+# Given a prefix, and a list of arguments, prefix the list of arguments and
+# assign to out: ie, addprefix(out "-I" "a;b;c") should set out to
+# "-Ia;-Ib;-Ic".
+function(addprefix out prefix) 
```

Comment:
Minor nit: While we haven't formalized or even really discussed coding standards, I have generally been trying to use underscores in function names to match the cmake coding style